Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
I would expect `optional` to return an optional honestly. The convention I've seen with iOS is to have the type you want there ... kinda like `UIColor.whiteColor().CGColor` so in this case I might expect `.Optional` to return an optional form. This function is `.optional` so it's lower case but I think it's close enough to be slightly confusing.